dftimewolf/lib/collectors/grr_hunt.py

```diff
@@ -247,20 +259,34 @@ def _extract_hunt_results(self, output_file_path):
     # Extract items from archive by host for processing
     collection_paths = []
     clients = set()
+    client_id_fqdn = {}
+    hunt_dir = None
     try:
       with zipfile.ZipFile(output_file_path) as archive:
         items = archive.infolist()
         for f in items:
+
+          if not hunt_dir:
+            hunt_dir = f.filename.split('/')[0]
+
+          # client_info.yaml contains information used to translate client ID
+          # to FQDN; extract it and save for later
+          if f.filename.split('/')[-1] == 'client_info.yaml':
+            client_id, fqdn = self._get_client_fqdn(archive.read(f))
+            client_id_fqdn[client_id] = fqdn
+            continue
+
+          # Otherwise, extract the files for each client ID in their own
+          # directory
           client_id = f.filename.split('/')[1]
           if client_id.startswith('C.'):
-            client_directory = os.path.join(self.output_path, client_id)
-            if not os.path.isdir(client_directory):
-              os.makedirs(client_directory)
             if client_id not in clients:
+              client_directory = os.path.join(self.output_path,
+                                              hunt_dir, client_id)
               collection_paths.append((client_id, client_directory))
               clients.add(client_id)
             try:
-              archive.extract(f, client_directory)
+              archive.extract(f, self.output_path)
```

Comment:
Why append a different path (L286) than you extract to?

Reply:
We want to append one directory per client so that this results in multiple timelines when sent to TimesketchExporter, but we want to extract everything in the same directory so as to preserve the GRR archive structure (and end up with something like `/tmp/<tmpdir>/<hunt_id>/<client_id>`). Extracting to an individual directory would've resulted in a directory like: `/tmp/<tmpdir>/<client_id>/<hunt_id>/<client_id>`.